src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs

```diff
@@ -7,171 +7,193 @@
         {
-            ContainingType = containingType;
-            IsOverride = isOverride;
-            TypeWithAnnotations = TypeWithAnnotations.Create(containingType.DeclaringCompilation.GetWellKnownType(WellKnownType.System_Type), NullableAnnotation.NotAnnotated);
-            var overriddenProperty = OverriddenProperty;
-            if (overriddenProperty is object)
-            {
-                TypeWithAnnotations = overriddenProperty.TypeWithAnnotations;
-            }
-            var getAccessorName = overriddenProperty?.GetMethod?.Name ??
-                SourcePropertyAccessorSymbol.GetAccessorName(PropertyName, getNotSet: true, isWinMdOutput: false /* unused for getters */);
-            GetMethod = new GetAccessorSymbol(this, getAccessorName);
         }
 
-        public override NamedTypeSymbol ContainingType { get; }
-
-        public override MethodSymbol GetMethod { get; }
-
-        public override MethodSymbol? SetMethod => null;
-
-        public override RefKind RefKind => RefKind.None;
-
-        public override TypeWithAnnotations TypeWithAnnotations { get; }
-
-        public override ImmutableArray<CustomModifier> RefCustomModifiers => ImmutableArray<CustomModifier>.Empty;
-
-        public override ImmutableArray<ParameterSymbol> Parameters => ImmutableArray<ParameterSymbol>.Empty;
-
-        public override bool IsIndexer => false;
-
         public override bool IsImplicitlyDeclared => true;
 
-        public override ImmutableArray<PropertySymbol> ExplicitInterfaceImplementations => ImmutableArray<PropertySymbol>.Empty;
-
-        public override Symbol ContainingSymbol => ContainingType;
-
-        public override ImmutableArray<Location> Locations => ContainingType.Locations;
-
         public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => ImmutableArray<SyntaxReference>.Empty;
 
-        public override Accessibility DeclaredAccessibility => Accessibility.Protected;
-
-        public override bool IsStatic => false;
-
-        public override bool IsVirtual => !IsOverride;
-
-        public override bool IsOverride { get; }
-
-        public override bool IsAbstract => false;
-
-        public override bool IsSealed => false;
+        IAttributeTargetSymbol IAttributeTargetSymbol.AttributesOwner => this;
 
-        public override bool IsExtern => false;
+        AttributeLocation IAttributeTargetSymbol.AllowedAttributeLocations => AttributeLocation.None;
 
-        internal override bool HasSpecialName => false;
+        AttributeLocation IAttributeTargetSymbol.DefaultAttributeLocation => AttributeLocation.None;
 
-        internal override CallingConvention CallingConvention => CallingConvention.HasThis;
+        protected override Location TypeLocation
+            => ContainingType.Locations[0];
 
-        internal override bool MustCallMethodsDirectly => false;
+        protected override SyntaxTokenList GetModifierTokens(SyntaxNode syntax)
+            => new SyntaxTokenList();
```

Comment:
nit: could we use a singleton rather than allocate?

Reply:
> nit: could we use a singleton rather than allocate?

No allocation here, it is a struct

---
In reply to: [452488746](https://github.com/dotnet/roslyn/pull/45831#discussion_r452488746) [](ancestors = 452488746)

Reply:
Ah, thanks

---
In reply to: [452489159](https://github.com/dotnet/roslyn/pull/45831#discussion_r452489159) [](ancestors = 452489159,452488746)